library/std/src/sys/sync/rwlock/futex.rs

```diff
@@ -116,6 +133,7 @@ impl RwLock {
                 }
             }
 
+            // FIXME shouldn't this be an assert?
             // Check for overflow.
             if has_reached_max_readers(state) {
```

Comment:
Should this be an `assert!` instead?

Reply:
As in `assert!(!has_reached_max_readers(state), "too many active read locks on RwLock")`? That sounds reasonable, clippy usually suggests this.